app/src/main/java/com/github/fribourgsdp/radio/CreateUserProfileActivity.kt

```diff
@@ -0,0 +1,48 @@
+package com.github.fribourgsdp.radio
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View
+import android.widget.Button
+import android.widget.EditText
+
+class CreateUserProfileActivity : AppCompatActivity() {
+    private lateinit var nameInput : EditText
+    private lateinit var saveButton : Button
+    private lateinit var user : User
+    private var db = FirestoreDatabase()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_create_user_profile)
+
+
+
+        nameInput = findViewById(R.id.nameInput)
+        saveButton = findViewById(R.id.saveButton)
+
+
+        prefillFields()
+        saveButton.setOnClickListener(saveButtonBehaviour())
+    }
+
+    private fun saveButtonBehaviour() : View.OnClickListener{
+        return View.OnClickListener {
+            val textValue = nameInput.text
+            user = User(textValue.toString())
+            db.setUser("testCreateUserProfile",user)
+            user.save(this)
```

Comment:
This may be a problem because it recreates a new user and will overwrite all information previously stored on the device in the case where a user has loaded his playlists for example.
A better solution would be to make a User's name property mutable, and use a try-catch here which would do:
try{
user.load
user.setName(textValue.toString())
}
catch {
what you have written now
}

Reply:
This activity will be deleted anyway. This activity was to test the features I was going to implement on the UserProfileActivity.